tools/fuzzer/expr_gen.h

```diff
@@ -42,6 +42,50 @@ enum class ExprKind : unsigned char {
 };
 constexpr size_t NUM_GEN_EXPR_KINDS = (size_t)ExprKind::EnumLast + 1;
 
+enum class TypeKind : unsigned char {
+  ScalarType,
+  TaggedType,
+  PointerType,
+  ReferenceType,
+  EnumLast = ReferenceType,
+};
+constexpr size_t NUM_GEN_TYPE_KINDS = (size_t)TypeKind::EnumLast + 1;
+
+class Weights {
+ private:
+  std::array<float, NUM_GEN_EXPR_KINDS> expr_weights_;
+  std::array<float, NUM_GEN_TYPE_KINDS> type_weights_;
+
+  using expr_iterator = decltype(expr_weights_)::iterator;
+  using expr_const_iterator = decltype(expr_weights_)::const_iterator;
+
+  using type_iterator = decltype(type_weights_)::iterator;
+  using type_const_iterator = decltype(type_weights_)::const_iterator;
+
+ public:
+  expr_iterator expr_begin() { return expr_weights_.begin(); }
+  expr_iterator expr_end() { return expr_weights_.end(); }
```

Comment:
I feel like operating on iterators is kinda clunky. Can you just expose the array itself instead? Then you can do range for when you don't need the index.